src/Microsoft.CodeQuality.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.cs

```diff
@@ -0,0 +1,357 @@
+                return false;
+            }
+            else if (constant == 1 &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.LessThanOrEqual &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.GreaterThan)
+            {
+                return false;
+            }
+            else if (constant > 1)
+            {
+                return false;
+            }
+
+            return IsCountMethodInvocation(binaryOperation.RightOperand, enumerableType);
+        }
+
+        /// <summary>
+        /// Tries the get an <see cref="int"/> constant from the <paramref name="operation"/>.
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        /// <param name="constant">The constant the <paramref name="operation"/> represents, if succeeded, or zero if <paramref name="operation"/> is not a constant.</param>
+        /// <returns><see langword="true" /> <paramref name="operation"/> is a constant, <see langword="false" /> otherwise.</returns>
+        public static bool TryGetInt32Constant(IOperation operation, out int constant)
+        {
+            constant = default;
+
+            if (operation.Type?.SpecialType != SpecialType.System_Int32)
+            {
+                // TODO: Should we be checking for other integer types?
+                return false;
+            }
+
+            var comparandValueOpt = operation.ConstantValue;
+
+            if (!comparandValueOpt.HasValue)
+            {
+                return false;
+            }
+
+            constant = (int)comparandValueOpt.Value;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Checks the is a <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})" />
+        /// or <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource}, Func{TSource, bool})" /> invocation.
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        /// <param name="enumerableType">Type of the enumerable.</param>
+        /// <returns><see langword="true" /> if XXXX, <see langword="false" /> otherwise.</returns>
+        private static bool IsCountMethodInvocation(IOperation operation, INamedTypeSymbol enumerableType)
+        {
+            if (operation.Kind != global::Microsoft.CodeAnalysis.OperationKind.Invocation ||
+                !(operation is global::Microsoft.CodeAnalysis.Operations.IInvocationOperation invocationExpression) ||
+                !invocationExpression.TargetMethod.ContainingSymbol.Equals(enumerableType) ||
+                !invocationExpression.TargetMethod.Name.Equals(DoNotUseCountWhenAnyCanBeUsedAnalyzer.CountMethodName, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            // TODO: Should I be doing more checks?
+
+            var parameterCountOFfset = operation.Language == LanguageNames.VisualBasic ? -1 : 0;
```

Comment:
We normally don't have such language specific checks in language agnostic layer. Also can you add a comment and explain why this is needed.

Reply:
Looking at the code below, personally I don't think you need any of it - if a method named `Count` is invoked on `enumerableType`, we can assume it is for count computation regardless of what parameters it has. We do this as almost all places where well known method invocation checks are performed.